src/main/java/com/commercetools/sync/categories/utils/CategoryAssetUpdateActionUtils.java

```diff
@@ -39,36 +39,18 @@
                                                             @Nonnull final AssetDraft newAsset,
                                                             @Nonnull final CategorySyncOptions syncOptions) {
 
-        final List<UpdateAction<Category>> updateActions = buildUpdateActionsFromOptionals(asList(
-            buildChangeAssetNameUpdateAction(oldAsset, newAsset),
-            buildSetAssetDescriptionUpdateAction(oldAsset, newAsset),
-            buildSetAssetTagsUpdateAction(oldAsset, newAsset),
-            buildSetAssetSourcesUpdateAction(oldAsset, newAsset)
-        ));
+        final List<UpdateAction<Category>> updateActions = filterEmptyOptionals(
+            asList(
+                buildChangeAssetNameUpdateAction(oldAsset, newAsset),
+                buildSetAssetDescriptionUpdateAction(oldAsset, newAsset),
+                buildSetAssetTagsUpdateAction(oldAsset, newAsset),
+                buildSetAssetSourcesUpdateAction(oldAsset, newAsset)
+            ));
 
```

Comment:
Maybe you can add this helper:

` 
final static <T> List<T> asNotEmptyList(Optional<T>... elements) {
        return filterEmptyOptionals(asList(elements));
 }
`

and rewrite this to

`
     final List<UpdateAction<Category>> updateActions = asNotEmptyList(
                buildChangeAssetNameUpdateAction(oldAsset, newAsset),
                buildSetAssetDescriptionUpdateAction(oldAsset, newAsset),
                buildSetAssetTagsUpdateAction(oldAsset, newAsset),
                buildSetAssetSourcesUpdateAction(oldAsset, newAsset)
            );
`

Reply:
Thanks for the idea @ahalberkamp It's great! :) 👍 

Here I add the new method: 0c26148
Here are the usages: fd6a48c